lambda-buffers-codegen/src/LambdaBuffers/Codegen/Rust/Print/Derive.hs

```diff
@@ -0,0 +1,259 @@
+module LambdaBuffers.Codegen.Rust.Print.Derive (printDeriveEqBase, printDeriveIsPlutusData, printDeriveJson, rsTraitImplPrinters) where
+
+import Control.Lens ((^.))
+import Data.Foldable (for_)
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Data.Set (Set)
+import Data.Set qualified as Set
+import LambdaBuffers.Codegen.LamVal.Eq (deriveEqImpl)
+import LambdaBuffers.Codegen.LamVal.Json (deriveFromJsonImpl, deriveToJsonImpl)
+import LambdaBuffers.Codegen.LamVal.MonadPrint qualified as LV
+import LambdaBuffers.Codegen.LamVal.PlutusData (deriveFromPlutusDataImpl, deriveToPlutusDataImpl)
+import LambdaBuffers.Codegen.Print qualified as Print
+import LambdaBuffers.Codegen.Rust.Print (MonadPrint)
+import LambdaBuffers.Codegen.Rust.Print.LamVal (printInstance)
+import LambdaBuffers.Codegen.Rust.Print.Refs qualified as RR
+import LambdaBuffers.Codegen.Rust.Print.Syntax qualified as R
+import LambdaBuffers.ProtoCompat qualified as PC
+import Prettyprinter (Doc, align, braces, colon, comma, encloseSep, hcat, indent, lparen, parens, rparen, space, vsep, (<+>))
+import Proto.Codegen qualified as P
+import Proto.Codegen_Fields qualified as P
+
+rsTraitImplPrinters ::
+  MonadPrint m =>
+  Map
+    R.QTraitName
+    ( PC.ModuleName ->
+      PC.TyDefs ->
+      (Doc ann -> Doc ann) ->
+      PC.Ty ->
+      m (Doc ann)
+    )
+rsTraitImplPrinters =
+  Map.fromList
+    [ (RR.partialEqTrait, printDerivePartialEqBase)
+    , (RR.eqTrait, printDeriveEqBase)
+    , (RR.isPlutusDataTrait, printDeriveIsPlutusData)
+    , (RR.jsonTrait, printDeriveJson)
+    ]
+eqTraitMethodName :: R.ValueName
+eqTraitMethodName = R.MkValueName "eq"
+
+eqTraitMethodArgs :: [(R.ValueName, R.QTyName)]
+eqTraitMethodArgs = [(R.MkValueName "self", R.qBuiltin R.MkTyName "Self"), (R.MkValueName "other", R.qBuiltin R.MkTyName "Self")]
+
+eqTraitMethodReturns :: R.QTyName
+eqTraitMethodReturns = R.qBuiltin R.MkTyName "bool"
+
+lvEqBuiltinsBase :: LV.PrintRead R.QValName
+lvEqBuiltinsBase = LV.MkPrintRead $ \(_ty, refName) ->
+  Map.lookup refName $
+    Map.fromList
+      [ ("eq", R.qForeignRef R.MkValueName "lbr-prelude" ["lamval"] "eq")
+      , ("and", R.qForeignRef R.MkValueName "lbr-prelude" ["lamval"] "and")
+      , ("true", R.qBuiltin R.MkValueName "true")
+      , ("false", R.qBuiltin R.MkValueName "false")
+      , ("PhantomData", R.qForeignRef R.MkValueName "std" ["marker"] "PhantomData")
```

Comment:
Why is this here?

Reply:
oh, that was a failed attempt, to inject PhantomData, I'll remove it